src/main/java/org/jenkinsci/plugins/githubreleaseperformer/GitHubReleasePerformer.java

```diff
@@ -0,0 +1,225 @@
+package org.jenkinsci.plugins.githubreleaseperformer;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.Util;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import net.sf.json.JSONObject;
+import org.apache.commons.io.IOUtils;
+import org.jenkinsci.plugins.tokenmacro.TokenMacro;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHReleaseBuilder;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+
+public class GitHubReleasePerformer extends Builder {
+
+	private final String apiUrl;
+	private final String tag;
+	private final String branch;
+	private final String releaseNotesFile;
+	private final String user;
+	private final String password;
+	private final String owner;
+	private final String repository;
+
+	private GitHub github;
+	private GHRepository ghRepository;
+
+	private static final String RELEASE_NOTES_TEMPLATE =
+			"### New features\n\n* [JIRA-TICKET](jira-link): Some feature\n\n### Improvements\n\n* [JIRA-TICKET](jira-link): Some improvement\n\n### Bug fixes\n\n* [JIRA-TICKET](jira-link): Some bug fix";
+
+	@SuppressWarnings("unused")
+	@DataBoundConstructor
+	public GitHubReleasePerformer(String apiUrl, String user, String password, String tag, String branch, String releaseNotesFile,
+			String owner,
+			String repository) {
+		this.apiUrl = apiUrl;
+		this.tag = Util.fixEmptyAndTrim(tag);
+		this.branch = Util.fixEmptyAndTrim(branch);
+		this.releaseNotesFile = Util.fixEmptyAndTrim(releaseNotesFile);
+		this.user = user;
+		this.password = password;
+		this.owner = Util.fixEmptyAndTrim(owner);
+		this.repository = Util.fixEmptyAndTrim(repository);
+	}
+
+	@Override
+	public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
+			BuildListener listener) {
+		String resolvedUser = user;
+		String resolvedPassword = password;
+		String resolvedRepository = repository;
+		String resolvedOwner = owner;
+		String resolvedTag = tag;
+		String resolvedBranch = branch;
+		String resolvedReleaseNotesFile = releaseNotesFile;
+		try {
+			resolvedUser = TokenMacro.expandAll(build, listener, user);
+			resolvedPassword = TokenMacro.expandAll(build, listener, password);
+			resolvedRepository = TokenMacro.expandAll(build, listener, repository);
+			resolvedOwner = TokenMacro.expandAll(build, listener, owner);
+			resolvedTag = TokenMacro.expandAll(build, listener, tag);
+			resolvedBranch = TokenMacro.expandAll(build, listener, branch);
+			resolvedReleaseNotesFile = TokenMacro.expandAll(build, listener, releaseNotesFile);
+		} catch (Exception e) {
+			e.printStackTrace(listener.error("Unable to resolve macros"));
```

Comment:
The message you pass to the listener is not very descriptive, should add at least some parameters so we could identify what's wrong when it goes wrong. Also why not just log through the listener but print the stacktrace to sout?